src/Facade/ResourceBuilders/MrReportResourceBuilder.cs

```diff
@@ -0,0 +1,227 @@
+                    "Week",
+                    null,
+                    detail.WeekAndYear,
+                    relativeWeek,
+                    detail.Segment);
+
+                this.SetValue(
+                    detailResources,
+                    "Ending",
+                    null,
+                    detail.WeekEnding,
+                    relativeWeek,
+                    detail.Segment);
+
+                this.SetValue(
+                    detailResources,
+                    "Stock",
+                    detail.Stock,
+                    null,
+                    relativeWeek,
+                    detail.Segment);
+
+                this.SetValue(
+                    detailResources,
+                    "Min Rail",
+                    detail.MinRail,
+                    null,
+                    relativeWeek,
+                    detail.Segment);
+
+                this.SetValue(
+                    detailResources,
+                    "Max Rail",
+                    detail.MaxRail,
+                    null,
+                    relativeWeek,
+                    detail.Segment);
+            }
+
+            return detailResources.OrderBy(a => a.Segment).ThenBy(b => b.DisplaySequence);
+        }
+
+        private void SetValue(
+            IEnumerable<MrDetailResource> detailResources,
+            string title,
+            decimal? value,
+            string textValue,
+            decimal relativeWeek,
+            int segment)
+        {
+            var detail = detailResources.FirstOrDefault(x => x.Title == title && x.Segment == segment);
+            if (detail == null)
+            {
+                return;
+            }
+
+            this.SetRelativeWeekValue(detail, relativeWeek, value, textValue);
+        }
+
+        private IEnumerable<MrDetailResource> CreateDetails(string title, int segments, int sequence)
+        {
+            for (int j = 0; j < segments; j++)
+            {
+                yield return new MrDetailResource { DisplaySequence = sequence, Segment = j, Title = title };
+            }
+        }
+
+        private IEnumerable<LinkResource> BuildHeaderLinks(MrHeader entity)
+        {
+            if (entity == null)
+            {
+                yield break;
+            }
+
+            yield return new LinkResource
+                             {
+                                 Rel = "part-used-on",
+                                 Href = $"/purchasing/material-requirements/used-on-report?partNumber={entity.PartNumber}"
+                             };
+            yield return new LinkResource { Rel = "part", Href = $"/parts/{entity.PartId}" };
+        }
+
+        private void SetRelativeWeekValue(MrDetailResource detail, decimal relativeWeek, decimal? value, string textValue)
+        {
+            var stringValue = string.IsNullOrEmpty(textValue) ? value.ToString() : textValue;
+            switch (relativeWeek)
+            {
+                case -1:
```

Comment:
this resource builder seems to be doing a lot of switching.. Could Unit test it? Or use a real one in the integration test and check it builds the right resource?

Reply:
yep good spot. it's on my list to add unit tests for it. i am torn with just doing real integration tests because i do think that has advantages but unit tests are easier for testing all the routes through the code

Reply:
yeah.. integration tests involve loads of boiler-platey overhead if you have loads of different cases. I think I'd go unit too